src/odbc_connection.cpp

```diff
@@ -1281,6 +1282,18 @@ class CallProcedureAsyncWorker : public ODBCAsyncWorker {
         strcat(combinedProcedureName, ".");
       }
       strcat(combinedProcedureName, (const char*)data->procedure);
+      #else
+      wchar_t *combinedProcedureName = new wchar_t[1024]();
+      if (data->catalog != NULL) {
+        wcscat(combinedProcedureName, data->catalog);
+        wcscat(combinedProcedureName, L".");
+      }
+      if (data->schema != NULL) {
+        wcscat(combinedProcedureName, data->schema);
+        wcscat(combinedProcedureName, L".");
+      }
+      wcscat(combinedProcedureName, data->procedure);
+      #endif
```

Comment:
strcat/wcscat is not really the most efficient way to do this. Would be more efficient, simpler, and more clear by using eg. sprintf. Something like std::fmt or https://fmt.dev would be even better, though.

Since we're just copying the narrow code path, this seems fine, but we should look at improvements for the future.

Reply:
With something like sprintf, I would still need the if-checks, correct?

Reply:
I was gonna say it shouldn't be too bad with just two printf calls, but I see now you have both catalog and schema. Conceivably this gives you 4 different printfs, but you could do it in one using ternaries:

```c
sprintf(combinedProcedureName, "%s%s%s%s%s",
  data->catalog ? data->catalog : "",
  data->catalog ? "." : "",
  data->schema ? data->schema : "",
  data->schema ? "." : "",
  data->procedure);

Alternatively, you can skip the printf and keep track of the end of line manually:
```c
auto str = combinedProcedureName;
if (data->catalog) {
  auto len = strlen(data->catalog);
  memcpy(str, data->catalog, len);
  str += len;
  *str = ".";
  str++;
}
if (data->schema) {
  auto len = strlen(data->schema);
  memcpy(str, data->schema, len);
  str += len;
  *str = ".";
  str++;
}
strcpy(str, data->procedure);
```